great_expectations/core/config_provider.py

```diff
@@ -0,0 +1,169 @@
+    """
+
+    def __init__(self, runtime_environment: Dict[str, str]) -> None:
+        self._runtime_environment = runtime_environment
+
+    def get_values(self) -> Dict[str, str]:
+        return self._runtime_environment
+
+
+class EnvironmentConfigurationProvider(AbstractConfigurationProvider):
+    """
+    Responsible for the management of environment variables.
+    """
+
+    def get_values(self) -> Dict[str, str]:
+        return dict(os.environ)
+
+
+class ConfigurationVariablesConfigurationProvider(AbstractConfigurationProvider):
+    """
+    Responsible for the management of user-defined configuration variables.
+
+    These can be found in the user's /uncommitted/config_variables.yml file.
+    """
+
+    def __init__(
+        self, config_variables_file_path: str, root_directory: Optional[str] = None
+    ) -> None:
+        self._config_variables_file_path = config_variables_file_path
+        self._root_directory = root_directory
+
+    def get_values(self) -> Dict[str, str]:
+        env_vars = dict(os.environ)
+        try:
+            # If the user specifies the config variable path with an environment variable, we want to substitute it
+            defined_path: str = substitute_config_variable(  # type: ignore[assignment]
+                self._config_variables_file_path, env_vars
+            )
+            if not os.path.isabs(defined_path):
+                root_directory: str = self._root_directory or os.curdir
+            else:
+                root_directory = ""
+
+            var_path = os.path.join(root_directory, defined_path)
+            with open(var_path) as config_variables_file:
+                contents = config_variables_file.read()
+
+            variables = dict(yaml.load(contents)) or {}
+            return cast(
+                Dict[str, str], substitute_all_config_variables(variables, env_vars)
+            )
+
+        except OSError as e:
+            if e.errno != errno.ENOENT:
+                raise
+            return {}
+
+
+class CloudConfigurationProvider(AbstractConfigurationProvider):
+    """
+    Responsible for the management of a user's GX Cloud credentials.
+
+    See `GeCloudConfig` for more information. Note that this is only registered on the primary
+    config provider when in a Cloud-backend environment.
+    """
+
+    def __init__(self, cloud_config: GeCloudConfig) -> None:
+        self._cloud_config = cloud_config
+
+    def get_values(self) -> Dict[str, str]:
+        from great_expectations.data_context.data_context.cloud_data_context import (
+            GECloudEnvironmentVariable,
+        )
+
+        return cast(
+            Dict[str, str],
+            {
+                GECloudEnvironmentVariable.BASE_URL: self._cloud_config.base_url,
+                GECloudEnvironmentVariable.ACCESS_TOKEN: self._cloud_config.access_token,
+                GECloudEnvironmentVariable.ORGANIZATION_ID: self._cloud_config.organization_id,
+            },
```

Comment:
Why do we need to cast here?

Reply:
it's because `org_id` is nullable for some reason - I think I can assert or cast that instead of this whole dict.

Reply:
Might be able to use a generic on the ABC here.

Reply:
Actually, could you ignore the type error here with a comment about `GECloudEnvironmentVariable.ORGANIZATION_ID`.

Reply:
done!